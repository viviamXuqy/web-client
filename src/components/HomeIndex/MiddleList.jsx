import React from 'react';
import { List, Button } from 'antd';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

import Cicon from '../../shared/components/Cicon';
import { AUTH } from '../../constants/site';
import Cookies from 'js-cookie';

let data = [
  {
    id: 1,
    title: 'App.header.tab.equipment',
    className: 'camera',
    path: '/cameras/ipc',
  },
  {
    id: 2,
    title: 'App.header.tab.task',
    className: 'mission',
    path: '/tasks/analysis',
  },
  {
    id: 3,
    title: 'App.header.tab.result',
    className: 'scan',
    path: '/results/resultsList',
  },
  {
    id: 4,
    title: 'App.header.tab.smartapp',
    className: 'apply',
    path: '/smartapp',
  },
  {
    id: 5,
    title: 'App.header.tab.system',
    className: 'system',
    path: '/system/setting',
  },
];

class MiddleList extends React.PureComponent {
  // 当组件输出到 DOM 后会执行 componentDidMount() 钩子
  componentDidMount() {
  }

  // 组件被从DOM中移除，React会调用componentWillUnmount()这个钩子函数
  componentWillUnmount() {
  }

  render() {
    Cookies.set('user', JSON.stringify({
      grade: 3,
    }));
    const user = JSON.parse(Cookies.get('user'));
    data = user && data.filter(item => AUTH[user.grade].includes(item.path));
    return (
      <div>
        <List
          className="homeindex-results-list"
          grid={{ gutter: 16, column: data.length }}
          dataSource={data}
          renderItem={item => (
            <List.Item>
              <Link to={item.path}>
                <div className="text-center">
                  <Button type="primary" shape="circle" icon="" size="large" >
                    <Cicon className="index-page-middle-icon" type={item.className} />
                  </Button>
                  <div className="index-page-middle-label"><FormattedMessage id={item.title} /></div>
                </div>
              </Link>
            </List.Item>
          )}
        />
      </div>
    );
  }
}

export default MiddleList;
