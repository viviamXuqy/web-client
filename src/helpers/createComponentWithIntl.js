import React from 'react';
import renderer from 'react-test-renderer'; // eslint-disable-line
import LocaleProvider from 'containers/Locale';
import { setStore } from '../utils/storage';
import { nodeWithIntlProp } from './intl-enzyme-test-helper';

const createComponentWithIntl = (children, props = { lang: 'zh-CN' }) => {
  setStore('locale', props.lang);
  return renderer.create(
    <LocaleProvider {...props}>
      {nodeWithIntlProp(children)}
    </LocaleProvider>,
  );
};

export default createComponentWithIntl;
