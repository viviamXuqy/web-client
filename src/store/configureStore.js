import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import api from '../middleware/api';
import error from '../middleware/error';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const isDev = process.env.NODE_ENV === 'development';
const loggerMiddleware = isDev ? [createLogger()] : [];

const configureStore = history => {
  const store = createStore(
    rootReducer,
    composeEnhancers(applyMiddleware(
      routerMiddleware(history),
      thunk, api, error, ...loggerMiddleware,
    )),
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../reducers', () => {
      const nextRootReducer = require('../reducers').default // eslint-disable-line
      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
};

export default configureStore;
