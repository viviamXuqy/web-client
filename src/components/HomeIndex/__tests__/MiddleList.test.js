import React from 'react';
import createComponentWithIntl from 'helpers/createComponentWithIntl';
import { mountWithIntl } from 'helpers/intl-enzyme-test-helper';
import { MemoryRouter } from 'react-router-dom';

import Cookies from 'js-cookie';

import MiddleList from 'components/HomeIndex/MiddleList';

import configureStore from '../../../store/__tests__/configureStore.test';

const { state } = configureStore;

const initProps = {
  lang: 'zh-CN',
};

function initializeData() {
  Cookies.set('user', JSON.stringify({grade:3 }));
  // Cookies.set('user', JSON.stringify(state.user));
}

const setup = (props = initProps) => {
  const component = <MiddleList {...props} />;

  const componentWithRouter = (
    <MemoryRouter>
      {component}
    </MemoryRouter>
  );

  const componentWithProvider = createComponentWithIntl(componentWithRouter);

  const wrapper = mountWithIntl(componentWithRouter);

  return {
    componentWithProvider,
    wrapper,
  };
};

describe('HomeIndex MiddleList component', () => {
  beforeEach(() => {
    initializeData();
  });

  it('should render without errors', () => {
    const { wrapper } = setup();

    expect(wrapper).toBeTruthy();
  });

  it('should render correct things', () => {
    const { wrapper } = setup();

    expect(wrapper.find('.text-center').length).toBe(5);
  });

  it("should render right thing and don't change unexpected", () => {
    const { componentWithProvider } = setup();
    const tree = componentWithProvider.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
