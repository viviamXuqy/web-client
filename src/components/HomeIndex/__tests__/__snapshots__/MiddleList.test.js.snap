// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeIndex MiddleList component should render right thing and don't change unexpected 1`] = `
<div>
  <div
    className="ant-list homeindex-results-list ant-list-split ant-list-grid"
  >
    <div
      className="ant-spin-nested-loading"
      style={null}
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-row"
          style={
            Object {
              "marginLeft": -8,
              "marginRight": -8,
            }
          }
        >
          <div
            className="ant-col-4"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <a
                  href="/cameras/ipc"
                  onClick={[Function]}
                >
                  <div
                    className="text-center"
                  >
                    <button
                      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="anticon iconfont icon-camera index-page-middle-icon"
                      />
                    </button>
                    <div
                      className="index-page-middle-label"
                    >
                      <span>
                        设备管理
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            className="ant-col-4"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <a
                  href="/tasks/analysis"
                  onClick={[Function]}
                >
                  <div
                    className="text-center"
                  >
                    <button
                      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="anticon iconfont icon-mission index-page-middle-icon"
                      />
                    </button>
                    <div
                      className="index-page-middle-label"
                    >
                      <span>
                        结构化任务
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            className="ant-col-4"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <a
                  href="/results/resultsList"
                  onClick={[Function]}
                >
                  <div
                    className="text-center"
                  >
                    <button
                      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="anticon iconfont icon-scan index-page-middle-icon"
                      />
                    </button>
                    <div
                      className="index-page-middle-label"
                    >
                      <span>
                        识别结果
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            className="ant-col-4"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <a
                  href="/smartapp"
                  onClick={[Function]}
                >
                  <div
                    className="text-center"
                  >
                    <button
                      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="anticon iconfont icon-apply index-page-middle-icon"
                      />
                    </button>
                    <div
                      className="index-page-middle-label"
                    >
                      <span>
                        智能应用
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            className="ant-col-4"
          >
            <div
              className="ant-list-item"
              style={
                Object {
                  "paddingLeft": 8,
                  "paddingRight": 8,
                }
              }
            >
              <div
                className="ant-list-item-content ant-list-item-content-single"
              >
                <a
                  href="/system/setting"
                  onClick={[Function]}
                >
                  <div
                    className="text-center"
                  >
                    <button
                      className="ant-btn ant-btn-primary ant-btn-circle ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <i
                        className="anticon iconfont icon-system index-page-middle-icon"
                      />
                    </button>
                    <div
                      className="index-page-middle-label"
                    >
                      <span>
                        系统管理
                      </span>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
